code/modules/lighting/light_effect.dm

```diff
@@ -0,0 +1,135 @@
+/atom/movable/light
+	mouse_opacity = 0
+	plane = LIGHTING_PLANE_MASTER
+
+	layer = 1
+	//layer 1 = base plane layer
+	//layer 2 = base shadow templates
+	//layer 3 = wall lighting overlays
+	//layer 4 = light falloff overlay
+
+	appearance_flags = KEEP_TOGETHER
+	icon = null
+	invisibility = INVISIBILITY_LIGHTING
+	pixel_x = -WORLD_ICON_SIZE*2
+	pixel_y = -WORLD_ICON_SIZE*2
+	glide_size = WORLD_ICON_SIZE
+	blend_mode = BLEND_ADD
+
+	// Prevent shadows from jerking over walls when walking with a flashlight.
+	animate_movement = 0
+
+	alpha = 180
+
+	var/current_power = 1
+	var/atom/movable/holder
+	var/point_angle
+	var/list/affecting_turfs = list()
+	var/list/temp_appearance
+
+/atom/movable/light/shadow
+	appearance_flags = KEEP_TOGETHER | TILE_BOUND
+	animate_movement = 1
```

Comment:
```suggestion
	animate_movement = SLIDE_STEPS
```
Note that this one is *not* just changing a value to its equivalent define. `SLIDE_STEPS` is `2` and is what nearly everything else in the codebase uses.
It may work better with `SYNC_STEPS`? I'm not really sure how that one works. Try it out and see which is better.